frontend/lib/resolution/Resolver.cpp

```diff
@@ -3760,21 +3772,22 @@ void Resolver::handleCallExpr(const uast::Call* call) {
   if (!skip) {
     QualifiedType receiverType = methodReceiverType();
 
-    // Calling initializer without qualifier, e.g., `init(a, b, c);``
-    //
     // If the user has mistakenly instantiated a field of the type before
     // calling ``init``, then the receiver type will either be fully or
     // partially instantiated. This will cause a failure to resolve the
     // ``init`` call, and result in a confusing and unhelpful error message.
+    // To resolve this problem, manually compute the fully-generic type that
+    // is being initialized and reset the receiver.
```

Comment:
I think it would be worth commenting here that the error for initializing a field before an `init` call will be emitted by the InitResolver upon `init` resolution later on; when I first saw this I had the question of "but if we're making it resolve correctly, how do we still emit the error?"